### DISCUSSION TITLE
Is use statement superfluous in MySqlConnector?

### DISCUSSION THREAD

**Discussion by Dachande663** [laravel/framework#44469] [Q&A]
When MySQL connects to MySQL via a PDO DSN, the dbname value is provided. However, on L27, Laravel executes a USE statement to select the datatbase again. This means all connections result in an extra statement been sent, which has a non-zero latency for any network-based architecture.

Is there a need for the USE statement? Removing it does not affect any tests as far as I can see.

https://github.com/laravel/framework/blob/03795ab20ecf0c9bca61737b9a5c2a353712df9e/src/Illuminate/Database/Connectors/MySqlConnector.php#L27

**Comment by coppolafab** [comment]
I think no. I've been using a modified MySqlConnector for years without issues.
See here https://github.com/laravel/framework/pull/43020

You can bind a new class to "db.connector.mysql" and override some methods, "connect" in particular.
